### DISCUSSION TITLE
posts/docker-compose-install/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#208] [Comments]
# posts/docker-compose-install/

Install Docker Engine, containerd, and Docker Compose

https://technotim.live/posts/docker-compose-install/

**Comment by CKMartens** [comment]
Hello Tim,

the instructions contain a small stumbling block when using Debian as the operating system. Then you have to replace ubuntu with debian in the entry of the repo:
echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null

Greetings

**Reply by timothystewart6** [reply]
Thanks!  I've added a link to the page if using Debian!

**Comment by TheColetrain** [comment]
I had to type `docker compose version` you have indicated just `docker compose` 
but who knows, maybe I did it wrong.

**Reply by CRK1918** [reply]
Many of us run Debian, may update it with many different Linux distro?

**Reply by timothystewart6** [reply]
There's a link to debian in the post!

**Comment by Swedorican-Design** [comment]
I try typing these commands in and I'm getting errors.  I mad a fresh ubuntu server install and began copy and pasting your commands but they are not working.  Maybe it's new updates.  I'm no linux pro.  But I'm trying to follow your videos on ssl on everything and the best game server but I'm stuck at the start on the below commands.

I tried to find a video that you mad to go along with this file and i cannot find it.

Thanks for your time.

# Add the repository to Apt sources:
echo \
  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
sudo apt-get update

**Reply by Swedorican-Design** [reply]
I figured out the problem I was copy and pasts one line at a time instead of all at one...
sorry super noob

**Comment by digictl** [comment]
Would love to see the implementation for Fedora derivative distro's such as RHEL. Even SUSE.

**Comment by JoJoTomaz** [comment]
Hi Tim,

I am having major issues trying to get docker to work. This is the error message I get when trying to run a docker container.

docker: Error response from daemon: AppArmor enabled on system but the docker-default profile could not be loaded: running `/usr/sbin/apparmor_parser apparmor_parser -Kr /var/lib/docker/tmp/docker-default1085072110` failed with output: apparmor_parser: Unable to replace "docker-default".  Permission denied; attempted to load a profile while confined?

any idea what I can do?

**Comment by SmilingJoe** [comment]
These instructions worked great for installing Docker and Docker Compose on my Linux Mint system - thanks Tim!
